Core/CharmDataModel.cpp

```diff
@@ -626,12 +626,16 @@ int CharmDataModel::activeEventCount() const
 EventIdList CharmDataModel::eventsThatStartInTimeFrame( const QDate& start,
                                                         const QDate& end ) const
 {
+    // do the comparisons in UTC, which is much faster as we only need to convert
+    // start and end date then
+    const QDateTime startUTC = QDateTime(start, QTime(0, 0, 0, 0)).toUTC();
+    const QDateTime endUTC =  QDateTime(end, QTime(23, 59, 59, 999)).toUTC();
```

Comment:
endUtc looks wrong, to make the check below be equivalent, it should be "end, QTime(0, 0, 0, 0)" (as it compared date() < end, not <=). Also there's a space too much after the "="